FoodOrder.TestApi/Controller/RestaurantTestController.cs

```diff
@@ -0,0 +1,136 @@
+﻿using AutoMapper;
+using FluentAssertions;
+using FoodOrderApi.Controllers;
+using FoodOrderApi.DataProvider;
+using FoodOrderApi.Mappings;
+using FoodOrderApi.Model.DTO;
+using FoodOrderApi.TestApi.Mockdata;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Moq;
+using System.Net;
+
+namespace FoodOrderApi.TestApi.Controller
+{
+    public class TestRestaurantController
+    {
+        private Mock<IDataProvider> restaurantService;
+        private IMapper mapperMock;
+        private Mock<ILogger<RestaurantController>> loggerMock;
+
+        public TestRestaurantController()
+        {
+            restaurantService = new Mock<IDataProvider>();
+            loggerMock = new Mock<ILogger<RestaurantController>>();
+            if (mapperMock == null)
+            {
+                var mappingConfig = new MapperConfiguration(mc =>
+                {
+                    mc.AddProfile(new AutoMapperProfiles());
+                });
+                IMapper mapper = mappingConfig.CreateMapper();
+                mapperMock = mapper;
+            }
+        }
+
+        [Theory]
+        [InlineData(1, 10)]
+        [InlineData(3, 2)]
+        public async Task Get_AllRestaurantPaginationWithValidIndex__ShouldReturn200Status(int pageNumber, int pageSize)
+        {
+            /// Arrange
+            restaurantService.Setup(_ => _.GetRestaurantPaged(pageNumber, pageSize)).ReturnsAsync(RestaurantMockData.GetRestaurantPaged(pageNumber, pageSize));
+            var sut = new RestaurantController(restaurantService.Object, mapperMock, loggerMock.Object);
```

Comment:
Instead of naming the variable as SUT, name the variable to the actual name of the SUT.

Reply:
Updated the variable name